### PR TITLE
feat(sqlsmith): Generate EquiJoin

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Add the 'user-facing changes' label if your PR contains changes that are visible to users (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#3369

### CODE REVIEW

File: src/tests/sqlsmith/src/relation.rs [hunk 1]
```diff
@@ -16,45 +16,114 @@ use rand::prelude::SliceRandom;
...
 impl<'a, R: Rng> SqlGenerator<'a, R> {
-    /// A relation specified in the FROM clause.
     pub(crate) fn gen_from_relation(&mut self) -> TableWithJoins {
-        match self.rng.gen_range(0..=9) {
-            0..=9 => self.gen_simple_table(),
-            // TODO: unreachable, should change to 9..=9,
-            // but currently it will cause panic due to some wrong assertions.
-            10..=10 => self.gen_subquery(),
-            _ => unreachable!(),
+        let (from_relation, _) = self.gen_from_relation_with_cols();
```
Comment (neverchanje) [inline src/tests/sqlsmith/src/relation.rs:35]: Why return the columns if you don't use?

**Comment by kwannoel** [comment]
Simplified the failing query to:
```sql
create table t1(v1 varchar);
create table t2(v2 varchar);
create table t3(v3 varchar);
SELECT * FROM t3, t1 JOIN t2 ON t1.v1 = t2.v2;
```

**Comment by kwannoel** [comment]
Error is during `LogicalPhase`.

```
    LogicalProject { exprs: [$1, $3, $5] }
      LogicalJoin { type: Inner, on: true, output_indices: all }
        LogicalScan { table: t3, columns: [_row_id, v3] }
        LogicalJoin { type: Inner, on: ($3 = $5), output_indices: all }
          LogicalScan { table: t1, columns: [_row_id, v1] }
          LogicalScan { table: t2, columns: [_row_id, v2] }
```

`JOIN ON` should be using `$1=$3` instead.

**Comment by kwannoel** [comment]
Known issue: https://github.com/singularity-data/risingwave/issues/2771

**Review by neverchanje — APPROVED** [review]
You can temporarily disable equi join if it's not well supported currently.

**Pull request merged**